app/models.py

```diff
@@ -388,6 +391,17 @@ def serialize(self, data=None):
             "countersignedAt": countersigned_at,
         }, **(data or {}))
 
+        if self.agreement_details and self.agreement_details.get('uploaderUserId'):
+            user = User.query.filter(
+                User.id == self.agreement_details.get('uploaderUserId')
+            ).first()
+
+            if user:
```

Comment:
Seems like, if we do _need_ that information (ie, we can't ignore it on the front-end), then it would make sense it would fail here rather than just having the frontend not find something it  expects to always be there.

I'll ask @laurenceberry.